### PR TITLE
KafkaReceiver

### PR DESCRIPTION
**Pull request by jasonholloway**
The app now sets up a single KafkaReceiver, listening on topics specified by GATHERER_TOPICS, and blurts out received messages to the console.

### CODE REVIEW

File: src/KafkaReceiver/index.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+const log = require('debug')('gatherer');
+const logError = require('debug')('gatherer:error');
+
+// const topicNameBuilder = require('./topicNameBuilder');
+
+// const config = require('../util/config');
+// const providers = config.get('providers');
+//const kafkaConfig = config.get('kafka');
+
+// const topicNames = providers.map(provider => topicNameBuilder(kafkaConfig.topicEnvironment, provider));
```
Comment (iancotterill) [inline src/KafkaReceiver/index.js]: Code comments...

File: src/KafkaReceiver/index.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+            if(!rawMessages.push(msg)) {
+                log('Pausing consumer');
+                consumer.pause();
+            }
+        });
+
+        consumer.on('error', error => {                
+            logError('Kafka Exception', error);
+            throw error; //?????
```
Comment (iancotterill) [inline src/KafkaReceiver/index.js]: more comments

File: src/KafkaReceiver/index.js [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+//const kafkaConfig = config.get('kafka');
+
+// const topicNames = providers.map(provider => topicNameBuilder(kafkaConfig.topicEnvironment, provider));
+
+let receiverId = 0;
+
+class KafkaReceiver {
+
+    constructor(conf) {
+        this.id = receiverId++; //uuid.v4();
```
Comment (iancotterill) [inline src/KafkaReceiver/index.js]: once more with the commented code...

Comment (jasonholloway) [inline src/KafkaReceiver/index.js]: thought you'd missed that one :)

**Pull request merged**